### ISSUE TITLE
Cannot warmstart when some configurations raised exceptions

### ISSUE THREAD

**Issue by arberzela** [automl/HpBandSter#32]
Consider warmstarting from the evaluations present in some `results.json` file and some of the sampled evaluations during optimization raised an error (e.g. `OutOfMemory` error) and the following line is appended to the `results.json` for this configuration and a `np.inf` loss is given to this configuration to mark it as a bad one:

`[[10, 0, 0], 38.0, {"finished": 1544192253.4073746, "started": 1544191887.3157759, "submitted": 1544191887.3152575}, null, "Traceback (most recent call last):\n  File \"/home/zelaa/HpBandSter/hpbandster/core/worker.py\", line 206, in start_computation\n    result = {'result': self.compute(*args, config_id=id, **kwargs),\n  File \"/home/zelaa/Thesis/bohb-darts/cond/workers/darts_worker.py\", line 35, in compute\n    darts_source=self.darts_path)\n  File \"/home/zelaa/Thesis/bohb-darts/cond/workers/helper.py\", line 81, in darts_cifar10\n    subprocess.check_call(\" \".join(bash_strings), shell=True)\n  File \"/home/zelaa/anaconda3/envs/pytorch-0.3.1-cu8/lib/python3.5/subprocess.py\", line 271, in check_call\n    raise CalledProcessError(retcode, cmd)\nsubprocess.CalledProcessError: Command 'cd /home/zelaa/Thesis/bohb-darts/cond/workers/lib/darts_space; python train.py --cutout --auxiliary --save /home/zelaa/Thesis/bohb-darts/cond/data/BOHB/run_4420/10_0_0 --epochs 38 --edge_normal_0 sep_conv_5x5' returned non-zero exit status 1\n"]`

When trying to warmstart [line 133](https://github.com/automl/HpBandSter/blob/master/hpbandster/core/base_iteration.py#L133) in `core/base_iteration.py` will raise a `TypeError` exception since `result['loss'] = None` for the aforementioned configuration. Adding `(not result['loss'] is None)` to the condition in [line 133](https://github.com/automl/HpBandSter/blob/master/hpbandster/core/base_iteration.py#L133) and capturing the exception in [line 292](https://github.com/automl/HpBandSter/blob/master/hpbandster/optimizers/config_generators/bohb.py#L292) in `optimizers/config_generators/bohb.py` as follow:

<pre>
try:
    loss = job.result["loss"] if np.isfinite(job.result["loss"]) else np.inf
except TypeError:
    loss = np.inf
</pre>

will remove the issue and BOHB will warmstart succesfully.

**Comment by sfalkner** [comment]
Good point. I think it would be better to actively check for None and catch that, would you agree?

**Comment by arberzela** [comment]
Hi Stefan,

I agree, what I wrote above was just a quick suboptimal remedy for this issue for the code I have been running. and to let you know where the errors are coming from :)

Cheers,
Arber
